core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosSchedulerUtils.scala

```diff
@@ -356,4 +374,233 @@ private[mesos] trait MesosSchedulerUtils extends Logging {
     sc.conf.getTimeAsSeconds("spark.mesos.rejectOfferDurationForReachedMaxCores", "120s")
   }
 
+  /**
+   * Checks executor ports if they are within some range of the offered list of ports ranges,
+   *
+   * @param sc the Spark Context
+   * @param ports the list of ports to check
+   * @return true if ports are within range false otherwise
+   */
+  protected def checkPorts(sc: SparkContext, ports: List[(Long, Long)]): Boolean = {
+
+    def checkIfInRange(port: Long, ps: List[(Long, Long)]): Boolean = {
+      ps.exists(r => r._1 <= port & r._2 >= port)
+    }
+
+    val portsToCheck = ManagedPorts.getPortValues(sc.conf)
+    val nonZeroPorts = portsToCheck.filter(_ != 0)
+    val withinRange = nonZeroPorts.forall(p => checkIfInRange(p, ports))
+    // make sure we have enough ports to allocate per offer
+    ports.map(r => r._2 - r._1 + 1).sum >= portsToCheck.size && withinRange
+  }
+
+  /**
+   * Partitions port resources.
+   *
+   * @param conf the spark config
+   * @param ports the ports offered
+   * @return resources left, port resources to be used and the list of assigned ports
+   */
+  def partitionPorts(
+      conf: SparkConf,
```

Comment:
I won't press the point, but I like to keep the general design principal that a function only knows about what it needs to know about.  Since it doesn't need to know about the whole `conf`, I'd rather just pass the ports in the client.   It also makes the function signature more meaningful.